src/connection/HttpConnection.ts

```diff
@@ -161,16 +161,43 @@ export default class HttpConnection extends BaseConnection {
         // if the response is compressed, we must handle it
         // as buffer for allowing decompression later
         let payload = isCompressed || isVectorTile ? new Array<Buffer>() : ''
-        const onData = isCompressed || isVectorTile
-          ? (chunk: Buffer) => { (payload as Buffer[]).push(chunk) }
-          : (chunk: string) => { payload = `${payload as string}${chunk}` }
+        const onData = isCompressed || isVectorTile ? onDataAsBuffer : onDataAsString
```

Comment:
Why we don't add the logics of these two functions `onDataAsBuffer` and `onDataAsString` in just a simple `else` branch in line [159](https://github.com/elastic/elastic-transport-js/blob/56e788086039ca8c9463a211a9016cf4e8037592/src/connection/HttpConnection.ts#L159)? In this way we can reuse the same approach used for existing `content-length` header. Moreover, we will reduce the size since the only  difference between the two functions `onDataAsBuffer` and `onDataAsString` is the limit check on `maxCompressedResponseSize` and `maxResponseSize`. WDYT?

Reply:
> Why we don't add the logics of these two functions onDataAsBuffer and onDataAsString in just a simple else branch in line [159](https://github.com/elastic/elastic-transport-js/blob/56e788086039ca8c9463a211a9016cf4e8037592/src/connection/HttpConnection.ts#L159)?

We can't because the two functions are called asynchronously by the response stream multiple times (https://github.com/elastic/elastic-transport-js/blob/56e788086039ca8c9463a211a9016cf4e8037592/src/connection/HttpConnection.ts#L221).
You could have a single function where you check if the response is compressed, but it wouldn't make much difference.

Reply:
@delvedor you right, I didn't see the usage in `response.on`.